libcaja-private/caja-directory-async.c

```diff
@@ -3850,7 +3850,12 @@ is_link_trusted (CajaFile *file,
         return TRUE;
     }
 
-    if (caja_file_can_execute (file) && caja_file_is_local (file))
+    if (caja_file_can_execute (file))
+    {
+        return TRUE;
+    }
```

Comment:
IMHO, you can actually delete these lines, since there is no need to check here whether the .desktop file has the eXecution bit. 
```diff
--- a/libcaja-private/caja-directory-async.c
+++ b/libcaja-private/caja-directory-async.c
@@ -3850,7 +3850,7 @@ is_link_trusted (CajaFile *file,
         return TRUE;
     }
 
-    if (caja_file_can_execute (file) && caja_file_is_local (file))
+    if (caja_file_is_local (file))
     {
         GFile *location;
 
```

Reply:
done